pingcap/tidb
# *: support `ALTER PARTITION` clause

### What problem does this PR solve?

It is part of `Defining placement rules in SQL` project. Depends on unmerged PR https://github.com/pingcap/parser/pull/931.

### What is changed and how it works?

It adds the corresponding DDL codes to https://github.com/pingcap/parser/pull/921. Now `ALTER PARTITION` will be parsed and executed.

Cancelling is not considered, so it defaults to the default behavior.

### Check List

Tests

- Unit test
- Integration test
- Manual test

Need a manual test, start `pd-server`, `tikv-server`, `tidb-server`. Then create a partitioned table, and test sql `ALTER TABLE t ALTER PARTITION p ADD PLACEMENT POLICY CONSTRAINTS='+zone=sh' ROLE=leader REPLICAS=3;`.

Then use `pd-ctl config placement-rules load` to dump rules. There should be a new rule.

### Release note

- support `ALTER PARTITION` clause
